### ISSUE TITLE
ClassNotFound for ReplicatedMap.remove()

### ISSUE THREAD

**Issue by neilstevenson** [hazelcast/hazelcast#14235]
Reproducer, for 3.11

```
import com.hazelcast.client.HazelcastClient;
import com.hazelcast.core.Hazelcast;
import com.hazelcast.core.HazelcastInstance;

public class Main {
	
	private static class World implements java.io.Serializable {
		private static final long serialVersionUID = 1L;
	}
	
    public static void main(String[] args) {
    	HazelcastInstance server = Hazelcast.newHazelcastInstance();
    	HazelcastInstance client1 = HazelcastClient.newHazelcastClient();
    	HazelcastInstance client2 = HazelcastClient.newHazelcastClient();
    	World world = new World();
    	client1.getReplicatedMap("etwas").put("hello", world);
    	System.out.println("Before get()");
    	System.out.println(client2.getReplicatedMap("etwas").get("hello"));
    	System.out.println("Before remove()");
    	System.out.println(client2.getReplicatedMap("etwas").remove("hello"));
    	System.out.println("Before shuwdown()");
    	client1.shutdown();
    	client2.shutdown();
    	server.shutdown();
    }
}
```

Note the `get()` works, but the `remove()` throws an exception.

See [StackOverflow ticket 53652395](https://stackoverflow.com/questions/53652395/hazelcast-classnotfounderror-on-dedicated-member-for-hazelclient-replicated-map)

**Comment by neilstevenson** [comment]
Wrong reproducer, will repost.

**Comment by neilstevenson** [comment]
The original reproducer does work, but you need another server, so too complicated.

Here's a simpler way.

Run the basic Hazelcast 3.11 server from [downloads](https://hazelcast.org/download/#imdg)

Run this client

```
import com.hazelcast.client.HazelcastClient;
import com.hazelcast.core.HazelcastInstance;

public class Main {
	
	private static class World implements java.io.Serializable {
		private static final long serialVersionUID = 1L;
	}
	
    public static void main(String[] args) {
    	HazelcastInstance client = HazelcastClient.newHazelcastClient();
    	World world = new World();
    	client.getReplicatedMap("etwas").put("hello", world);
    	System.out.println(client.getReplicatedMap("etwas").get("hello"));
    	client.shutdown();
    }
}

```

**Comment by ahmetmircik** [comment]
closed by https://github.com/hazelcast/hazelcast/pull/14238

**Comment by ahmetmircik** [comment]
Tested again but could not replicate the issue with 3.11.1. 
With 3.11, i can replicate it. I used the code of stackoverflow question. Maybe you left server in memory format as OBJECT?

**Comment by neilstevenson** [comment]
Steps to recreate.
(1) Download 3.11.1 from https://hazelcast.org/download/#imdg
(2) Run `$ Downloads/hazelcast-3.11.1/bin/start.sh `
(3) Use
```
    public static void main(String[] args) throws InterruptedException {
    	HazelcastInstance client = HazelcastClient.newHazelcastClient();
    	World world = new World();
    	client.getReplicatedMap("etwas").put("hello", world);
    	System.out.println(client.getReplicatedMap("etwas").get("hello"));
    	client.shutdown();
    }
```

Gives
```
INFO: [192.168.1.193]:5701 [dev] [3.11.1] Received auth from Connection[id=1, /127.0.0.1:5701->/127.0.0.1:49711, endpoint=null, alive=true, type=JAVA_CLIENT], successfully authenticated, principal: ClientPrincipal{uuid='0f40bce3-fc23-446f-8b62-2b15ee004027', ownerUuid='b0eb497b-10b5-4281-8da4-d7ac48b06d94'}, owner connection: true, client version: 3.11.1
Dec 19, 2018 8:17:39 AM com.hazelcast.internal.partition.impl.PartitionStateManager
INFO: [192.168.1.193]:5701 [dev] [3.11.1] Initializing cluster partition table arrangement...
Dec 19, 2018 8:17:40 AM com.hazelcast.replicatedmap.impl.operation.PutOperation
SEVERE: [192.168.1.193]:5701 [dev] [3.11.1] java.lang.ClassNotFoundException: com.hazelcast.Main$World
com.hazelcast.nio.serialization.HazelcastSerializationException: java.lang.ClassNotFoundException: com.hazelcast.Main$World
	at com.hazelcast.internal.serialization.impl.JavaDefaultSerializers$JavaSerializer.read(JavaDefaultSerializers.java:86)
```

**Comment by ahmetmircik** [comment]
The exception is expected in this case. Because, by default, you have OBJECT in-memory-format on server side replicated-map. This means all values will be converted to OBJECT on server and you have to put the `World` class on server too.

The issue, that i think it was fixed in 3.11.1, appears on BINARY in-memory-formatted replicated-map. For BINARY replicated-map, you shouldn't need to put `World` class on server.

Your test should set server side memory format to BINARY. When you set it, do you still see the exception?

**Comment by neilstevenson** [comment]
Indeed! - https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/resources/hazelcast-default.xml#L281

User error, I hadn't released `OBJECT` was the default. `BINARY` would be consistent with IMap but I can see the sense in `OBJECT` for ReplicatedMap.
